src/screens/MiningServer.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { t } from 'ttag';
+import {
+  colors,
+} from '../constants';
+import LOCAL_STORE from '../storage';
+import ReactLoading from 'react-loading';
+import { getGlobalWallet } from "../modules/wallet";
+import { useDispatch, useSelector } from 'react-redux';
+import BackButton from '../components/BackButton';
+import HathorAlert from '../components/HathorAlert';
+import { updateMiningServer } from "../actions";
+
+/**
+ * Screen to change the tx-mining server that the wallet uses to mine transactions
+ *
+ * @memberof Screens
+ */
+function MiningServer() {
+  const wallet = getGlobalWallet();
+  const dispatch  = useDispatch();
+  const currentServer = useSelector(state => state.miningServer);
+
+  /* errorMessage {string} Message to be shown in case of error in form */
+  const [errorMessage, setErrorMessage] = useState('');
+  /* isHardwareWallet {boolean} If the application is using a hardware wallet */
+  const isHardwareWallet = LOCAL_STORE.isHardwareWallet();
+  /* loading {boolean} If should show spinner while waiting for server response */
+  const [loading, setLoading] = useState(false);
+
+  // Declare refs
+  const newServerRef = useRef(null);
+  const pinRef = useRef(null);
+  const alertSuccessRef = useRef();
+
+  useEffect(() => {
+    if (loading) {
+      alertSuccessRef.current.show(3000);
+    }
+    setLoading(false);
+  }, [currentServer]);
+
+  /**
+   * Called after user click the button to change the server
+   */
+  const changeServer = async (reset) => {
+    let newServer;
+
+    if (reset) {
+      if (currentServer === null) {
+        setErrorMessage(t`Already using the default server`);
+        return;
+      }
+    } else {
+      if (newServerRef.current.value === '') {
+        setErrorMessage(t`New server is not valid`);
+        return;
+      }
+
+      newServer = newServerRef.current.value;
+
+      if (newServer === currentServer) {
+        setErrorMessage(t`New server is the same as the current server`);
+        return;
+      }
+    }
+
+    // we don't ask for the pin on the hardware wallet
+    if (!isHardwareWallet) {
+      if (!await wallet.checkPin(pinRef.current.value)) {
+        setErrorMessage(t`Invalid PIN`);
+        return;
+      }
+    }
+
+    setErrorMessage('');
+    newServerRef.current.value = '';
+    pinRef.current.value = '';
+
+    setLoading(true);
+    dispatch(updateMiningServer(newServer, reset));
```

Comment:
This operation doesn't have anything async. Why do you need a loading?

Reply:
While there is no network or io call, the saga is an async call.
Alsoo, in case some error on the saga happens that does not update the current mining server the user will be left with an indicator that the change has not happened.

What do you think?

Reply:
If we're adding a loading, I think it should be in the saga (and remove it from the useEffect)

The component should just reflect what is in the redux store

In the case you're describing (an error in the saga), the loading will be there forever, because the `currentServer` will never change thus never causing the `useEffect` to run, right?